File: C++/026.cpp
// 026 Remove Duplicates From Sorted Array
//
//


class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if (nums.empty()) return 0;
        int count = 0;
        int n = nums.size();
        for (int i = 1; i < n; i++) {
            if (nums[i] == nums[i-1]) cnt++;
            else nums[i-cnt] = nums[i];
        }
        return n-cnt;
    }
};

File: Go/026.go
func removeDuplicates(nums[] int) int {
	for i := 0; i+1 < len(nums) {
		if nums[i] == nums[i+1] {
			nums = append(nums[:i], nums[i+1:]...)
		} else {
			i++
		}
	}
	return len(nums)
}
